aces/Loris
# [Doc Repo] Refactor to use new File Uploader class

## Brief summary of changes

* Use the new File Uploader class to do file uploading.
* Add a new permission `document_repository_edit` to separate view and edit permissions.
* Cleanup `files` class in the doc repo module

#### Testing instructions (if applicable)

1. Run the patch.
2. In the Configuration module under Paths, set `documentRepositoryPath` to a writable folder on your system.
2. Using a superuser, or another user with the **new** `document_repository_edit` permission, ensure that you can upload files.
3. Make sure that you can download the files you've uploaded.
4. Try uploading a file type that's not allowed, such as an `.mp3` file. You should get a Sweet Alert error message.
5. Remove the `document_repository_edit` permission. You should get a 403 response when trying to upload any file.

#### Links to related tickets (GitHub, Redmine, ...)

* #4181 
* Resolves #4854
* Resolves #6373
